src/test/java/com/exasol/adapter/dialects/scalarfunction/ScalarFunctionsAbstractIT.java

```diff
@@ -346,14 +346,67 @@ void testSimpleArithmeticFunctions(final ScalarFunctionCapability function, fina
         }
 
         @Test
-        void testCast() {
+        void testCast() throws SQLException {
             if (isTestDisabled(CAST))
                 return;
+            this.valueSetter.setValueOfSingleRow(2);
             runOnExasol(statement -> {
                 final String virtualSchemaQuery = this.queryBuilder
                         .buildQueryFor("CAST(" + this.columnName + " AS VARCHAR(254) UTF8)");
                 try (final ResultSet result = statement.executeQuery(virtualSchemaQuery)) {
-                    assertThat(result, table().row("2").matches(TypeMatchMode.STRICT));
+                    assertThat(result, table().row(String.valueOf(2)).matches(TypeMatchMode.STRICT));
+                }
+            });
+        }
+
+        @ParameterizedTest
+        @CsvSource({ "-2, 0", "2, 2" })
+        void testGreatest(final int input, final int expectedOutput) throws SQLException {
+            if (isTestDisabled(GREATEST))
+                return;
+            this.valueSetter.setValueOfSingleRow(input);
+            runOnExasol(statement -> {
+                final String virtualSchemaQuery = this.queryBuilder
+                        .buildQueryFor("GREATEST(" + this.columnName + ", 0)");
+                try (final ResultSet result = statement.executeQuery(virtualSchemaQuery)) {
+                    assertThat(result, table().row(expectedOutput).matches(TypeMatchMode.NO_JAVA_TYPE_CHECK));
+                }
+            });
+        }
+    }
+
+    @Nested
+    @TestInstance(PER_CLASS)
+    @Tag("WithDoubleParameter") // workaround since IDEA can find the test otherwise
+    class WithDoubleParameter {
+
+        private ScalarFunctionQueryBuilder queryBuilder;
+        private String columnName;
+        private VirtualSchemaTestTable.ValueSetter<Double> valueSetter;
+
+        @BeforeAll
+        void beforeAll() throws SQLException {
+            final VirtualSchemaTestTable<Double> testTable = createDoubleVirtualSchemaTable();
+            this.queryBuilder = new ScalarFunctionQueryBuilder(testTable.getFullyQualifiedName());
+            this.columnName = getColumnsOfTable(testTable.getFullyQualifiedName()).get(0);
+            this.valueSetter = testTable.getValueSetter();
+        }
+
+        @AfterAll
+        void afterAll() {
+            deleteCreatedTables();
+        }
+
+        @ParameterizedTest
+        @CsvSource({ "0.1, 0", "0.5, 1", "0.9, 1" })
+        void testRound(final double input, final double expectedOutput) throws SQLException {
+            if (isTestDisabled(ROUND))
```

Comment:
I would prefer a more meaningful name, reflecting the intent. Right now `isTestDisabled` only says that the test is not enabled but it doesn't say why.

Maybe we can change it to something like `isScalarFunctionSupported(ROUND))` ?

Reply:
I don't have more information in that class. It just gives a dialect the possibility to disable certain tests. If it is not supported, one should better remove the capability.